Source/Diagnostics/ParticleIO.cpp

```diff
@@ -299,3 +301,217 @@ storePhiOnParticles ( PinnedMemoryParticleContainer& tmp,
         }
     }
 }
+
+namespace
+{
+// This template is isolated in a separate namespace to simplify the usage of
+// the doDirectGatherVectorField template and avoid branching
```

Comment:
I would suggest changing the description to something like the following:

```suggestion
// This template function is called with all the possible combinations 
// of the template parameters by storeEMFieldsOnParticles.
// depos_order and galerkin_interpolation must be template parameters
// because they are template parameters of doDirectGatherVectorField,
// which is called by this function.
```